store/mockstore/unistore/tikv/mvcc.go

```diff
@@ -935,16 +935,16 @@ func (store *MVCCStore) buildPrewriteLock(reqCtx *requestCtx, m *kvrpcpb.Mutatio
 		lock.Op = uint8(kvrpcpb.Op_Put)
 	}
 	if rowcodec.IsRowKey(m.Key) && lock.Op == uint8(kvrpcpb.Op_Put) {
-		if rowcodec.IsNewFormat(m.Value) {
-			reqCtx.buf = m.Value
-		} else {
+		if !rowcodec.IsNewFormat(m.Value) {
 			reqCtx.buf, err = encodeFromOldRow(m.Value, reqCtx.buf)
 			if err != nil {
 				log.Error("encode data failed", zap.Binary("value", m.Value), zap.Binary("key", m.Key), zap.Stringer("op", m.Op), zap.Error(err))
 				return nil, err
 			}
+
+			lock.Value = make([]byte, len(reqCtx.buf))
+			copy(lock.Value, reqCtx.buf)
```

Comment:
I finally find the bug! Reuse the `reqCtx.buf` could make the `lock.Value` be overwritten by another row.

It seems that this bug requires 1PC, and set multiple rows in one transaction (if these rows are in different tables, it will panic), so this problem only occurs in the TTL test, in which we set the session variables to enable 1PC explicitly.

It takes me a lot of time to reach here :cry: .